File: src/main/java/com/indiepost/validation/ContentType.java
package com.indiepost.validation;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.*;

@Documented
@Retention(RetentionPolicy.RUNTIME)
@Constraint(validatedBy = {ContentTypeMultipartFileValidator.class})
@Target({ElementType.METHOD, ElementType.FIELD, ElementType.ANNOTATION_TYPE, ElementType.CONSTRUCTOR, ElementType.PARAMETER})
public @interface ContentType {

    String message() default "Default Error Message";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};

    /**
     * 허용되는 콘텐츠 타입
     * <p/>
     * 예:
     * <ul>
     * <li>application/pdf</li>
     * <li>application/msword</li>
     * <li>images/png</li>
     * </ul>
     *
     * @return 허용되는 콘텐츠 타입
     */
    String[] value();
}

File: src/main/kotlin/com/indiepost/validation/ContentType.kt
package com.indiepost.validation

import javax.validation.Constraint
import javax.validation.Payload
import kotlin.reflect.KClass

@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [ContentTypeMultipartFileValidator::class])
@Target(AnnotationTarget.FUNCTION,
        AnnotationTarget.PROPERTY_GETTER,
        AnnotationTarget.PROPERTY_SETTER,
        AnnotationTarget.FIELD,
        AnnotationTarget.ANNOTATION_CLASS,
        AnnotationTarget.CONSTRUCTOR,
        AnnotationTarget.VALUE_PARAMETER)
annotation class ContentType(
        /**
         * 허용되는 콘텐츠 타입
         *
         *
         * 예:
         *
         *  * application/pdf
         *  * application/msword
         *  * images/png
         *
         *
         * @return 허용되는 콘텐츠 타입
         */
        vararg val value: String,
        val message: String = "Default Error Message",
        val groups: Array<KClass<*>> = [],
        val payload: Array<KClass<out Payload>> = [])